lib/spack/spack/spec.py

```diff
@@ -4793,6 +4793,16 @@ def __init__(self, parse_error):
         self.string = parse_error.string
         self.pos = parse_error.pos
 
+    @property
```

Comment:
Where does this get used (or is the PR still in progress?)

Reply:
`long_message` is an attribute on `spack.error.SpackError`, so this is just changing how that long message is generated for this subclass.